### DISCUSSION TITLE
Process dependencies

### DISCUSSION THREAD

**Discussion by GauBen** [Hexagon/pup#31] [Ideas]
Hey! I'm considering using pup for a rather large project (~ 20 services), but a few features are currently lacking. One of the most important addition to pup before making the switch is the possibility to declare dependencies:

```jsonc
{
  "processes": [
    {
      "id": "api",
      "cmd": "yarn workspace api run dev"
    },
    {
      "id": "frontend",
      "cmd": "yarn workspace frontend run dev",
      "dependsOn": ["api"]
    }
  ]
}
```

With this config file, running `pup start frontend` would run `api` first.

Later on, if Pup features a healthcheck, the `dependsOn` field could be extended with features similar to [docker compose #depends_on](https://docs.docker.com/compose/compose-file/05-services/#depends_on):

```jsonc
{
  "id": "frontend",
  "dependsOn": {
    "api": { "condition": "healthy", "restart": false }
  }
}
```

**Comment by Hexagon** [comment]
I've though about this, but never really figured out a good way to implement it. This sounds lika a great way though. Are you interested in implementing a feature like this?

Note that pup supports similar shell operators as deno task shell, so some dependency logic could be carried out simply using &&, || and so on.

**Reply by GauBen** [reply]
I'd love to implement this, but it depends entirely on my company. I'll keep you updated if I get some time to work on Pup!
